amqp-impl/src/main/java/io/streamnative/pulsar/handlers/amqp/ExchangeContainer.java

```diff
@@ -57,18 +59,87 @@ public static void putExchange(NamespaceName namespaceName, String exchangeName,
+    /**
+     * Get or create exchange.
+     *
+     * @param namespaceName namespace used in pulsar
+     * @param exchangeName name of exchange
+     * @param createIfMissing true to create the exchange if not existed, and exchangeType should be not null
+     *                        false to get the exchange and return null if not existed
+     * @param exchangeType type of exchange: direct,fanout,topic and headers
+     * @return the completableFuture of get result
+     */
+    public CompletableFuture<AmqpExchange> asyncGetExchange(NamespaceName namespaceName,
+                                                            String exchangeName,
+                                                            boolean createIfMissing,
+                                                            String exchangeType) {
+        CompletableFuture<AmqpExchange> amqpExchangeCompletableFuture = new CompletableFuture<>();
+        if (StringUtils.isEmpty(exchangeType) && createIfMissing) {
+            log.error("exchangeType should be set when createIfMissing is true");
+            amqpExchangeCompletableFuture.complete(null);
+            return amqpExchangeCompletableFuture;
+        }
         if (namespaceName == null || StringUtils.isEmpty(exchangeName)) {
-            return null;
+            log.error("Parameter error, namespaceName or exchangeName is empty.");
+            amqpExchangeCompletableFuture.complete(null);
+            return amqpExchangeCompletableFuture;
+        }
+        if (pulsarService.getState() != PulsarService.State.Started) {
+            log.error("Pulsar service not started.");
+            amqpExchangeCompletableFuture.completeExceptionally(new PulsarServerException("PulsarService not start"));
+            return amqpExchangeCompletableFuture;
         }
         Map<String, AmqpExchange> map = exchangeMap.getOrDefault(namespaceName, null);
-        if (map == null) {
-            return null;
+        if (map == null || map.getOrDefault(exchangeName, null) == null) {
+            // check pulsar topic
+            String topicName = PersistentExchange.getExchangeTopicName(namespaceName, exchangeName);
+            executor.execute(() -> {
+                CompletableFuture<Topic> topicCompletableFuture =
+                        amqpTopicManager.getTopic(topicName, createIfMissing);
+                topicCompletableFuture.whenComplete((topic, throwable) -> {
+                    if (throwable != null) {
+                        log.error("Get topic error:{}", throwable.getMessage());
+                        amqpExchangeCompletableFuture.complete(null);
+                    } else {
+                        if (null == topic) {
+                            log.error("The exchange topic did not exist. namespace{}, exchangeName: {}",
+                                    namespaceName.toString(), exchangeName);
+                            amqpExchangeCompletableFuture.complete(null);
```

Comment:
If createIfMissing is false, we just want to find the existed topic, if the topic not existed, we should return null. This can be used in the condition of declare a passive exchnage or queue.

Reply:
Ok, make sense.